src/screenfull.js

```diff
@@ -109,9 +109,11 @@
 					resolve();
 				}.bind(this);
 
-				document[fn.exitFullscreen]();
-
-				this.on('change', onFullScreenExit);
+				if (document[fn.fullscreenElement]) {
+					document[fn.exitFullscreen]();
+					this.on('change', onFullScreenExit);
+				}
+				resolve();
```

Comment:
This is still not correct. When in fullscreen and exciting, the promise will resolve before it's actually excited.

Reply:
Oh yes, it's returning a promise, should be fixed now.